libraries/lib-time-and-pitch/StaffPad/TimeAndPitch.cpp

```diff
@@ -354,6 +360,34 @@ void TimeAndPitch::_process_hop(int hop_a, int hop_s)
     for (int ch = 0; ch < _numChannels; ++ch)
       vo::calcPhases(d->spectrum.getPtr(ch), d->phase.getPtr(ch), d->spectrum.getNumSamples());
 
+    if (_shiftTimbreCb)
+       _shiftTimbreCb(
+          1 / _pitchFactor, d->spectrum.getPtr(0), d->norm.getPtr(0));
+
+    if (_reduceImaging && _pitchFactor < 1.)
+    {
+       // Upsampling brings spectral components down, including those that were
+       // beyond the Nyquist. From `imagingBeginBin`, we have a mirroring of the
+       // spectrum, which, if not lowpass-filtered by the resampler, may yield
+       // an aliasing-like quality if what is mirrored is a harmonic series. A
+       // simple thing to do against that is to scramble the phase values there.
+       // This way we preserve natural energy fluctuations, but the artifact
+       // isn't noticeable as much anymore, because much more noise-like, which
+       // is what one typically gets at those frequencies.
+       const int imagingBeginBin = fftSize / 2 * _pitchFactor + 1;
+       const auto n = _numBins - imagingBeginBin;
+       auto pSpec = d->spectrum.getPtr(0);
+       std::transform(
```

Comment:
I understand that this is a new code, but I will stress that it is located in a known hot point

Reply:
Yes, but I cannot think of a more optimized alternative, and there are extenuating circumstances:
* only gets executed for clips that are pitch-shifted downwards,
* for a number of bins that is at most a quarter of the FFT size, when down-shifted by an octave, and less for lesser shift amounts.

Reply:
Consider me convinced:-)